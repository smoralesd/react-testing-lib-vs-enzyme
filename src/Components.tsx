import * as React from "react";

export type TopProps = MiddleProps & {
  topMessage: string;
};

export class Top extends React.PureComponent<TopProps> {
  public render() {
    return (
      <>
        <h1>Container</h1>
        Container message: {this.props.topMessage}
        <Middle {...this.props} />
      </>
    );
  }
}

type MiddleProps = BottomProps & {
  middleMessage: string;
};

class Middle extends React.PureComponent<MiddleProps> {
  public render() {
    return (
      <>
        <h2>In the middle</h2>
        Middle message is {this.props.middleMessage}
        <br></br>
        <Bottom {...this.props} />
      </>
    );
  }
}

type BottomProps = {
  bottomProps: string;
};

class Bottom extends React.PureComponent<BottomProps> {
  public render() {
    return <span>Leaf message is: {this.props.bottomProps}</span>;
  }
}
