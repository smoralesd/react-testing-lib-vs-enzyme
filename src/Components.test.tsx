import { render } from '@testing-library/react';
import { shallow } from "enzyme";
import * as React from 'react';
import { Top, TopProps } from "./Components";

describe("Component", () => {
  const props: TopProps = {
    topMessage: "top level container message",
    middleMessage: "middle tier message",
    bottomProps: "bottom level message",
  };

  it('renders using enzyme and matches snapshot', () => {
      const result = shallow(<Top {...props}/>);

      expect(result.getElement()).toMatchSnapshot();
  });

  it('renders using react-test-library and matches snapshot', () => {
      const result = render(<Top {...props}/>);

      expect(result).toMatchSnapshot();
  });
});
