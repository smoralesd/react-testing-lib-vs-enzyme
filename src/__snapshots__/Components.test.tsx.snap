// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders using enzyme and matches snapshot 1`] = `
<React.Fragment>
  <h1>
    Container
  </h1>
  Container message: 
  top level container message
  <InTheMiddle
    containerMessage="top level container message"
    inTheMiddleMessage="middle tier message"
    leafMessage="bottom level message"
  />
</React.Fragment>
`;

exports[`Component renders using react-test-library and matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1>
        Container
      </h1>
      Container message: 
      top level container message
      <h2>
        In the middle
      </h2>
      Middle message is 
      middle tier message
      <br />
      <span>
        Leaf message is: 
        bottom level message
      </span>
    </div>
  </body>,
  "container": <div>
    <h1>
      Container
    </h1>
    Container message: 
    top level container message
    <h2>
      In the middle
    </h2>
    Middle message is 
    middle tier message
    <br />
    <span>
      Leaf message is: 
      bottom level message
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
